contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/DrillHiveMetaStoreClient.java

```diff
@@ -209,8 +213,10 @@ private DrillHiveMetaStoreClient(final HiveConf hiveConf) throws MetaException {
       throws TException {
     try {
       return mClient.getAllTables(dbName);
+    } catch (MetaException | UnknownDBException e) {
+      throw e;
     } catch (TException e) {
-      logger.warn("Failure while attempting to get hive tables", e);
+      logger.warn("Failure while attempting to get hive tables. Retries once.", e);
```

Comment:
Thanks for the suggestion. I modified the code by calling close() explicitly. However, to make it work, we also have to change the client from nonClosable to Closable(), by removing the overriden close() method.